### PR TITLE
IN-894 Update project configuration and linting defaults

### PR DESCRIPTION
**Pull request by jonavellecuerdo**
### What does this PR do?

Update the Python CLI template to include the latest (and greatest 🎉 ) of the DataEng team's standards for Python project code linting, formatting, and security checks. We recently revisited our current practices and made the following key decisions:

### Helpful background context

The team has created some Confluence documentation around these decisions:

* [Python Projects Linters](https://mitlibraries.atlassian.net/l/cp/2fMbkuvg)
* [Python Projects Pre-Commit](https://mitlibraries.atlassian.net/l/cp/ppHaRimC)
* `.dockerignore` file: I essentially copied the contents of the `.gitignore` file. My assumption was that many of the files we don't want to commit to Github are also files that we don't want to include in a Docker container (e.g. cache files, .env files).

### How can a reviewer manually see the effects of these changes?

1. Clone this repo.
2. Check into this branch.
3. Run `make install`
4. Run make lint
5. Run make test

### Includes new or updated dependencies?

YES

### What are the relevant tickets?

https://mitlibraries.atlassian.net/browse/IN-894

### Developer

- [x] All new ENV is documented in README (or there is none)
- [ ] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,9 +1,6 @@
...
-
 import pytest
 
 
 @pytest.fixture(autouse=True)
-def test_env():
-    os.environ = {"WORKSPACE": "test"}
-    yield
+def _test_env(monkeypatch):
+    monkeypatch.setenv("WORKSPACE", "test")
```
Comment (ghukill) [inline tests/conftest.py:6]: +1 use of `monkeypatch`

**Review by ghukill — APPROVED** [review]
Looks good to me!  Thanks for updating these templates.  

I know we still have some open questions about configuring python projects, and maybe there are some extremely low level integration tests we could add (e.g. ping the deployed lambda).   But I'd vote to move these agreed upon changes through, and tackle those separately.

File: lambdas/my_function.py [hunk 2]
```diff
@@ -23,9 +23,10 @@
     logger.info("No Sentry DSN found, exceptions will not be sent to Sentry")
 
 
-def lambda_handler(event: dict, context: object) -> str:  # noqa
+def lambda_handler(event: dict) -> str:
     if not os.getenv("WORKSPACE"):
-        raise RuntimeError("Required env variable WORKSPACE is not set")
+        unset_workspace_error_message = "Required env variable WORKSPACE is not set"
```
Comment (ehanson8) [inline lambdas/my_function.py:28]: Is it worth creating a var here? The name isn't much shorter than the actual message so I don't see the value in changing this

**Review by ehanson8 — COMMENTED** [review]
1 suggestion

File: lambdas/my_function.py (hunk 2 above)
Comment (jonavellecuerdo) [reply lambdas/my_function.py:28]: Hmm, the original format threw this Ruff error: https://docs.astral.sh/ruff/rules/raise-vanilla-args/. I took this from `alma-webhook-lambdas`, but the other solution they propose is creating a custom Exception class. What do you think is best? 🤔

Comment (ehanson8) [reply lambdas/my_function.py:28]: Oh, didn't realize that caused a linting error, don't worry about it!

**Review by ehanson8 — APPROVED** [review]

**Pull request merged**